gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -324,15 +337,17 @@ def create_per_sample_counts_ht(
     :param filter_group_ht: Table containing filter groups for summary stats.
     :return: Table containing per-sample variant counts.
     """
-    # Add extra Allele Count and Allele Type annotations to variant MatrixTable,
-    # according to Hail standards, to help their computation.
-    variant_ac, variant_types = vmt_sample_qc_variant_annotations(
+    # Add Allele Type annotations to variant MatrixTable.
+    _, variant_types = vmt_sample_qc_variant_annotations(
```

Comment:
yeah, I thought about that too, but was hoping it happens small enough times per sample that we will still get a good per sample variant count @ch-kr and @mike-w-wilson do you think it's worth the effort to add this in now vs adding a note that it wasn't done, and will have a small impact on per sample numbers?

Reply:
For frequency this is how we do it, but we very much shouldn't densify for these stats, so it would have to be some sort of modification. 

Also, how to handle sex chromosomes...

This just gets more and more complicated!

```
def densify_and_prep_vds_for_freq(
    vds: hl.vds.VariantDataset,
    ab_cutoff: float = 0.9,
) -> hl.MatrixTable:
    """
    Densify VDS and select necessary annotations for frequency and histogram calculations.

    Select entry annotations required for downstream work. 'DP', 'GQ', and '_het_ab' are
    required for histograms. 'GT' and 'adj' are required for frequency calculations.
    '_high_ab_het_ref' is required for high AB call corrections in frequency and
    histogram annotations.

    Assumes all necessary annotations are present:
        - adj
        - _het_non_ref
        - GT
        - GQ
        - AD
        - DP
        - sex_karyotype

    :param vds: Input VDS.
    :param ab_cutoff: Allele balance cutoff to use for high AB het annotation.
    :return: Dense MatrixTable with only necessary entry annotations.
    """
    logger.info("Densifying VDS...")
    mt = hl.vds.to_dense_mt(vds)

    logger.info("Computing sex adjusted genotypes...")
    ab_expr = mt.AD[1] / mt.DP
    gt_expr = adjusted_sex_ploidy_expr(mt.locus, mt.GT, mt.sex_karyotype)

    mt = mt.select_entries(
        "DP",
        "GQ",
        GT=gt_expr,
        adj=get_adj_expr(gt_expr, mt.GQ, mt.DP, mt.AD),
        _het_ab=ab_expr,
        _high_ab_het_ref=(ab_expr > ab_cutoff) & ~mt._het_non_ref,
    )
    return mt
```

Reply:
I vote we just add a note -- I think the effort to payoff ratio is unbalanced for rerunning after the fixing het nonref issue. what do you mean about the sex chromosomes?

Reply:
Just that as is, we are not adjusting for ploidy either, so I probably need to add that in also.
I switched to have an option for only autosomes when I was still trying to get some numbers for Kaitlin, but I should probably add support for sex chromosomes. It's on this thread because I thought about it when I was looking over the code for the high allele balance het issue.

Reply:
Sorry I missed this Friday. Only 198772 sites (0.06% of all sites) have any high AB hets so I agree with a note.